gmsm/gmsm-htb.js

```diff
@@ -0,0 +1,504 @@
+                    sessionId: sessionId,
+                    partnerId: __profile.partnerId,
+                    adm: bidCreative,
+                    requestId: returnParcel.requestId,
+                    size: returnParcel.size,
+                    price: targetingCpm,
+                    dealId: bidDealId,
+                    timeOfExpiry: __profile.features.demandExpiry.enabled ? __profile.features.demandExpiry.value
+                        + System.now() : 0
+                });
+
+                //? if(FEATURES.INTERNAL_RENDER) {
+                returnParcel.targeting.pubKitAdId = pubKitAdId;
+                //? }
+            }
+        }
+
+        if (!bidReceived) {
+            //? if (DEBUG) {
+            Scribe.info(__profile.partnerId + ' returned no demand for placement: '
+                + returnParcel.xSlotRef.placementId);
+            //? }
+            returnParcel.pass = true;
+        }
+
+        if (__profile.enabledAnalytics.requestTime) {
+            var result = 'hs_slot_pass';
+            if (bidReceived) {
+                result = 'hs_slot_bid';
+            }
+            EventsService.emit(result, headerStatsInfo);
+        }
+    }
+
+    /* =====================================
+     * Constructors
+     * ---------------------------------- */
+
+    (function __constructor() {
+        EventsService = SpaceCamp.services.EventsService;
+        RenderService = SpaceCamp.services.RenderService;
+        ComplianceService = SpaceCamp.services.ComplianceService;
+
+        __profile = {
+            partnerId: 'GmsmHtb',
+            namespace: 'GmsmHtb',
+            statsId: 'GMSM',
+            version: '2.0.0',
+            targetingType: 'slot',
+            enabledAnalytics: {
+                requestTime: true
+            },
+            features: {
+                demandExpiry: {
+                    enabled: false,
+                    value: 0
+                },
+                rateLimiting: {
+                    enabled: false,
+                    value: 0
+                }
+            },
+            targetingKeys: {
+                id: 'ix_gmsm_id',
+                om: 'ix_gmsm_om',
+                pm: 'ix_gmsm_dealid'
+            },
+            bidUnitInCents: 0.01,
+            lineItemType: Constants.LineItemTypes.ID_AND_SIZE,
+            callbackType: Partner.CallbackTypes.ID,
+            architecture: Partner.Architectures.MRA,
+            requestType: Partner.RequestTypes.ANY
+        };
+
+        /* --------------------------------------------------------------------------------------- */
+
+        //? if (DEBUG) {
+        var PartnerSpecificValidator = GmsmValidator;
+
+        var results = ConfigValidators.partnerBaseConfig(configs) || PartnerSpecificValidator(configs);
+
+        if (results) {
+            throw Whoopsie('INVALID_CONFIG', results);
+        }
+        //? }
+
+        __baseUrl = Browser.getProtocol() + '//secure.adnxs.com/jpt';
```

Comment:
In preparation for the cookie policy changes coming to all modern browsers, we are recommending that all requests be sent as HTTPS from now on.